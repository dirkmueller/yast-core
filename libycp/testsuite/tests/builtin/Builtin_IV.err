Parsed:
----------------------------------------------------------------------
(toset ([1, 5, 3, 2, 3, true, false, true]) == [false, true, 1, 2, 3, 5])
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
(setcontains ([1, 2, 5], 2) == true)
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
(tolower ("aBcDeF") == "abcdef")
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
(toascii ("a\326B\304c") == "aBc")
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
(toascii ("Special characters: !#$%&()*+,-./:;<=>?@[]^_`{}|~") == "Special characters: !#$%&()*+,-./:;<=>?@[]^_`{}|~")
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
(flatten ([[1, 2], [3, 4]]) == [1, 2, 3, 4])
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
(flatten ([[1, [2, [3, [4]]]]]) == [1, [2, [3, [4]]]])
----------------------------------------------------------------------
[Parser] tests/builtin/Builtin_IV.ycp:67 No match for 'flatten : <unspec> (list)'
[Parser] tests/builtin/Builtin_IV.ycp:67 Please fix parameter types to match one of:
[Parser] tests/builtin/Builtin_IV.ycp:67 'flatten : list <<T>> (const list <list <<T>>>)'
[Parser] tests/builtin/Builtin_IV.ycp:67 Wrong parameters in call to flatten(...)
[Parser] tests/builtin/Builtin_IV.ycp:67 Expected '(const list <list <<T>>>)', seen '(list)'.
runycp: parser error
