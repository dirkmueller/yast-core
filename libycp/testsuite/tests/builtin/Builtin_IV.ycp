
# ---------------------------------------------------------
#
#  Filename:    Builtin_IV.ycp
#
#  Purpose:     test cases for YCP test
#
#  Creator:     jd@suse.de
#
#  Maintainer:  jd@suse.de
#
# ---------------------------------------------------------

# ---------------- added 1999/11/29 ---------------

//
//   toset(list l) -> list
//
// Scans a list for duplicates, removes them and sorts the list.
//

( toset([1,5,3,2,3,true,false,true]) == [false,true,1,2,3,5] )




//
//     setcontains(list l, any v) -> boolean
//
// Determines, if a certain value <tt>v</tt> is contained in
// a list <tt>l</tt>, but assumes that <tt>l</tt> is sorted. If <tt>l</tt> is
// not sorted, the result is undefined.
//

( setcontains([1,2,5], 2) == true )

//
//    tolower( string s ) -> string
//
// Returns a string that results from string<tt>s</tt> by
// converting each character tolower.


( tolower( "aBcDeF" ) == "abcdef" )


//
//    toascii( string s ) -> string
//
// Returns a string that results from string<tt>s</tt> by
// copying each character that is below 0x7F (127).
//
( toascii( "aÖBÄc" ) == "aBc" )
( toascii( "Special characters: !#$%&()*+,-./:;<=>?@[]^_`{}|~") == "Special characters: !#$%&()*+,-./:;<=>?@[]^_`{}|~" )



//
//     flatten(list(list(type)) l) -> list(type)
//
// Gets a list l of lists and creates a single list that is 
// the concatenation of all lists in l.
//

( flatten([ [1,2], [3,4] ]) == [1, 2, 3, 4] )
( flatten([ [1,[2,[3,[4]]]] ] ) == [1, [2, [3, [4]]]] )
( flatten([ [1], 2, [3,4] ]) == [1, 2, 3, 4] )

