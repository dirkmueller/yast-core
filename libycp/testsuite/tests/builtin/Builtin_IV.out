(true)
(true)
(true)
(true)
(true)
(true)
(true)
(nil)
